tests/test_auth/TestControllers.py

```diff
@@ -117,3 +118,59 @@ def test_github_token_validity(self):
         """
         res = github_token_validity('token')
         self.assertEqual(res, False)
+
+class ManageAccount(BaseTestCase):
+
+    def test_edit_username(self):
+        """
+        Tests if username is edited
+        """
+        self.create_user_with_role(
+            self.user.name, self.user.email, self.user.password, Role.admin)
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/account/manage',data=dict(
+                    current_password=self.user.password,
+                    name="T1duS",
+                    email=self.user.email
+                ))
+            user = User.query.filter(User.name == "T1duS").first()
+            self.assertNotEqual(user, None)
+
+    def test_edit_email(self):
+        """
+        Tests if user email is edited
+        """
+        self.create_user_with_role(
+            self.user.name, self.user.email, self.user.password, Role.admin)
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/account/manage',data=dict(
+                    current_password=self.user.password,
+                    name=self.user.name,
+                    email="valid@gmail.com"
+                ))
+            user = User.query.filter(User.email == "valid@gmail.com").first()
+            self.assertNotEqual(user, None)
+
+    def test_edit_invalid_email(self):
+        """
+        Test for editing an invalid email
+        """
+        self.create_user_with_role(
+            self.user.name, self.user.email, self.user.password, Role.admin)
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/account/manage',data=dict(
+                    current_password=self.user.password,
+                    name=self.user.name,
+                    email="invalid@gg"
+                ))
+            user = User.query.filter(User.email == "invalid@gg").first()
+            self.assertEqual(user, None)                   
```

Comment:
How is this different from the test above `test_edit_email` apart from diff email?

Reply:
this is assertEqual, the other was assertNotEqual

Reply:
@T1duS Makes sense. Sorry, I misread it. Still would like to have assert for testing if the failure message is in response?